docs/tutorials/hn_data_prep.mdx

```diff
@@ -0,0 +1,378 @@
+---
+title: "Data Ingestion and Model-Assisted Labeling with xorq"
+description: "Learn how to fetch HackerNews data and automatically label data with sentiment using OpenAI GPT models"
+---
+
+This is part 1/4.
+
+## Overview
+
+In this tutorial, you'll learn how to:
+- Set up xorq and configure the necessary components
+- Fetch data from the [HackerNews API](https://github.com/HackerNews/API)
+- Use OpenAI gpt-3.5-turbo model to automatically label data with sentiment analysis
+- Create a labeled dataset ready for future machine learning tasks
+
+## Prerequisites
+
+<Accordion title="Prerequisites">
+- Python 3.8+ installed on your system
+- An OpenAI API key for the sentiment labeling
+- Basic understanding of Python and data processing pipelines
+</Accordion>
+
+<Warning>
+Make sure to set your OpenAI API key in your environment:
+```bash
+export OPENAI_API_KEY=your_api_key
+```
+</Warning>
+
+## Installation and Imports
+
+First, install xorq and the required dependencies:
+
+```bash
+pip install xorq pandas 
+```
+
+Then import the necessary modules:
+
+```python
+import pandas as pd
+import xorq as xo
+import xorq.expr.datatypes as dt
+
+from xorq.caching import ParquetStorage
+from xorq.common.utils.import_utils import import_python
+
+m = import_python(xo.options.pins.get_path("hackernews_lib"))
+o = import_python(xo.options.pins.get_path("openai_lib"))
+```
+
+<Info>
+The imported modules `m` (hackernews_lib) and `o` (openai_lib) contain utility
+functions for:
+- Connecting to the HackerNews Firebase API
+- Fetching and processing HackerNews stories
+- Making calls to OpenAI's API for sentiment analysis
+- Processing the response into structured data
+
+You'll need to ensure these files are accessible in your environment or create
+them based on the code snippets in the Appendix.
+</Info>
+
+## Defining the HackerNews Fetcher
+
+We'll define a User-Defined Expression Function (UDXF) that fetches HackerNews stories:
```

Comment:
So **UDXF** stands for User-Defined Expression Function, should we add a definition in the **Core Concepts**?

Reply:
This should User-Defined Exchange Function. I will update.